core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java

```diff
@@ -446,190 +402,153 @@ public void close() {
 
   @Override
   public boolean namespaceExists(Namespace namespace) {
+    if (exists(JdbcUtil.GET_NAMESPACE_SQL, catalogName, JdbcUtil.namespaceToString(namespace) + "%")) {
+      return true;
+    }
+
+    if (exists(JdbcUtil.GET_NAMESPACE_PROPERTIES_SQL, catalogName, JdbcUtil.namespaceToString(namespace) + "%")) {
+      return true;
+    }
+
+    return false;
+  }
+
+  private int execute(String sql, String... args) {
+    return execute(err -> {}, sql, args);
+  }
+
+  private int execute(Consumer<SQLException> sqlErrorHandler, String sql, String... args) {
```

Comment:
Error handler seems unnecessary since it's never used and throws anyway (as opposed to delegating to the handler).

Reply:
This is used in one case: https://github.com/apache/iceberg/pull/4231/files#diff-a5f7966c7493d463facbf7e76fe0fa3db1b637d29ee12e5b3628a5e5dd289e0aR195

I did this to avoid having a SQLException try/catch in every method.

Reply:
Makes sense.  Missed that usage.